### DISCUSSION TITLE
Memory cache for app icons using resource uri

### DISCUSSION THREAD

**Discussion by kawaiiDango** [coil-kt/coil#911] [Q&A] [answered]
**Describe the bug**
I am using https://github.com/coil-kt/coil/issues/79#issuecomment-533903394 to load app icons in a recyclerview.
Memory cache doesn't seem to work as I see the placeholder image and the crossfade every time I scroll back.

**Expected behavior**
Memory cache should work

**To Reproduce**
How can we reproduce this? Please attach a small project that reproduces the bug.

**Logs/Screenshots**
If applicable, add logs or screenshots to help explain your problem.

**Version**
1.3.2

**Comment by colinrtwhite** [comment, marked as answer]
Is your image not eligible to be cached in the memory cache (i.e. is it an animated image)? A project that reproduces the issue would help.

**Reply by kawaiiDango** [reply]
The result is AdaptiveIconDrawable. I am using the code
```
val packageName = "com.android.settings" // Replace with the correct app's package name.
val appInfo = packageManager.getApplicationInfo(packageName, 0)
imageView.load("android.resource://$packageName/${appInfo.icon}")
```

**Reply by colinrtwhite** [reply]
Yep you'll need to convert that to a `BitmapDrawable`. You can do this with any transformation added to your request. This should work:

```kotlin
class BitmapTransformation() {
    fun cacheKey() = "BitmapTransformation"
    suspend fun transform(input: Bitmap, size: Size) = input
}
```
